ilc/server/GuardManager.js

```diff
@@ -0,0 +1,40 @@
+const actionTypes = require('../common/guard/actionTypes');
+
+class GuardManager {
+    #transitionHooksPlugin;
+
+    constructor(pluginManager) {
+        this.#transitionHooksPlugin = pluginManager.getTransitionHooksPlugin();
+    }
+
+    async hasAccessTo(req, res) {
+        let hasAccess = true;
+
+        if (this.#transitionHooksPlugin === null) {
+            return hasAccess;
+        }
+
+        const route = req.raw.router.getRoute();
+        const hooks = this.#transitionHooksPlugin.getTransitionHooks();
+
+        if (hooks.length === 0) {
+            return hasAccess;
+        }
+
+        const actions = await Promise.all(hooks.map((hook) => hook({
```

Comment:
Some try/catch with explicit error is necessary here...

Reply:
Updated